### PR TITLE
Add tox

### PR DESCRIPTION
**Pull request by sfdye**
- [x] Migrate all virtualenvs to tox 
- [x] Support for python3.5 - Python 3.8
- [x] Add travis check for flake8/black/isort
- [x] Add a tox command `format` for manual formatting

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[tox]
+envlist = py{36,37,38},flake8
+
+[testenv]
+deps = -rtest-requirements.txt
+commands =
+	python setup.py test
```
Comment (s-t-e-v-e-n-k) [inline tox.ini]: This is the deprecated command that tox is supposed to be getting us away from. :-)

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[tox]
+envlist = py{36,37,38},flake8
```
Comment (s-t-e-v-e-n-k) [inline tox.ini]: This is missing py35.

File: .travis.yml [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+  include:
+    - python: 3.5
+      env: TOXENV=py35
+    - python: 3.6
+      env: TOXENV=py36
+    - python: 3.7
+      env: TOXENV=py37
+    - python: 3.8
+      env: TOXENV=py38
+    - env: TOXENV=flake8
```
Comment (s-t-e-v-e-n-k) [inline .travis.yml]: This to be reads like we run flake8 under 3.8, but tox.ini says 3.6. Can we pick one?

File: .travis.yml [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
 install:
-  - pip install codecov flake8
+  - pip install tox
 script:
-  - flake8
-  - coverage run --source github,tests setup.py test
```
Comment (s-t-e-v-e-n-k) [inline .travis.yml]: Does this mean we lose coverage reports?

Comment (sfdye) [reply .travis.yml]: Thanks for the feedback. Again this PR is specifically in Draft mode, which means it's not yet ready for review

Comment (s-t-e-v-e-n-k) [reply .travis.yml]: It's never early for reviews! :-)

Comment (sfdye) [reply .travis.yml]: The draft mode was mean to not disturb the reviewers 😬

File: tox.ini (hunk 1 above)
Comment (sfdye) [reply tox.ini]: we might need to migrate to pytest first, but that will be a separate PR

File: .flake8 [hunk 5]
```diff
@@ -1,4 +0,0 @@
-[flake8]
```
Comment (sfdye) [inline .flake8]: this is moved to tox.ini

**Comment by sfdye** [comment]
@s-t-e-v-e-n-k Now it's ready for review :)

File: test-requirements.txt [hunk 6]
```diff
@@ -1,3 +1,7 @@
 cryptography
 httpretty>=0.9.6
 parameterized>=0.7.0
+coverage==5.0.3
+black==19.10b0; python_version > '3.5'
```
Comment (s-t-e-v-e-n-k) [inline test-requirements.txt]: I don't think we need the marker here, we require 3.5 or greater anyway

File: .travis.yml [hunk 7]
```diff
@@ -1,14 +1,19 @@
...
+  include:
+    - python: 3.5
+      env: TOXENV=py35
+    - python: 3.6
+      env: TOXENV=py36
+    - python: 3.7
+      env: TOXENV=py37
+    - python: 3.8
+      env: TOXENV=py38
+    - env: TOXENV=lint
```
Comment (s-t-e-v-e-n-k) [inline .travis.yml]: I remain concerned about which Python version Travis will use when tox is configured for 3.6, but if it works, then that's great too.

Comment (sfdye) [reply .travis.yml]: It works. I set the base python to py36 for lint.

File: test-requirements.txt (hunk 6 above)
Comment (sfdye) [reply test-requirements.txt]: This is because black is only available for py3.6+

**Comment by sfdye** [comment]
image

Now travis will fail either flake8/isort/black fails the check.

**Pull request merged**